### PR TITLE
Add pyproject.toml and drop support for python < 3.8

### LINKED ISSUE #99: Add pyproject.toml for build requirements
Probably would also drop some older versions of python at the same time tbh.

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #99 

Hopefully this covers it but would love a look over from @IAlibay

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[build-system]
+# Minimum requirements for the build system to execute
+requires = [
+    "cython>=0.28,<3.0",
+    "packaging",
```
Comment (IAlibay) [inline pyproject.toml:5]: Are you directly using packaging somewhere for version checks?

**Comment by IAlibay** [comment]
I'd suggest also adding a pure pip wheel build (see the MDA azure pipelines stuff for an example)

File: pyproject.toml (hunk 1 above)
Comment (hmacdope) [reply pyproject.toml:5]: No don't think so I'll pull it

**Comment by hmacdope** [comment]
@IAlibay Hopefully this covers the pip only build. I'll do a release because forgot to make tags on master (#100) and then we can proceed on the MDA side.

File: setup.py [hunk 2]
```diff
@@ -8,8 +8,8 @@
 from setuptools import setup, Command, Extension
 
 # Make sure I have the right Python version.
-if sys.version_info[:2] < (3, 6):
-    print('MDAnalysis requires Python 3.6 or better. Python {0:d}.{1:d} detected'.format(*
+if sys.version_info[:2] < (3, 8):
+    print('MDAnalysis requires Python 3.8 or better. Python {0:d}.{1:d} detected'.format(*
           sys.version_info[:2]))
     print('Please upgrade your version of Python.')
     sys.exit(-1)
```
[the remark this answers is not in the document]
Comment (IAlibay) [reply setup.py:15]: Also do you not have any install time dependencies? I'm not seeing an `install_requires` anywhere?

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple more things

File: setup.py [hunk 3]
```diff
@@ -7,13 +7,6 @@
 
 from setuptools import setup, Command, Extension
 
-# Make sure I have the right Python version.
```
Comment (IAlibay) [inline setup.py:10]: Whilst you're here, I think you can safely assume that `print_function` isn't going to be a problem (if someone does try to use py2.7 they'll get a bit more gibberish, but at this point it'll be gibberish either way I think).

**Review by IAlibay — APPROVED** [review]

**Pull request merged**